tartufo/scanner.py

```diff
@@ -462,21 +462,32 @@ def chunks(self) -> Generator[types.Chunk, None, None]:
         already_searched: Set[bytes] = set()
 
         try:
-            if self.git_options.branch:
-                branches = self._repo.remotes.origin.fetch(self.git_options.branch)
+            if self.git_options.local_only:
+                if self.git_options.branch:
+                    unfiltered_branches = list(self._repo.branches)
+                    branches = [
+                        x
+                        for x in unfiltered_branches
+                        if x.name == self.git_options.branch
+                    ]
+                else:
+                    branches = self._repo.branches
             else:
-                branches = self._repo.remotes.origin.fetch()
+                if self.git_options.branch:
+                    branches = self._repo.remotes.origin.fetch(self.git_options.branch)
```

Comment:
Near as I can tell, this is returning a list of Head objects that point to remote, rather than just the list of fetched Heads locally.  Thus I would expect the scan is taking place on the remote branches rather than the local fetched copies.  Otherwise for an already fetched repo the performance difference wouldn't really exist.